web/src/js/components/Header/FlowMenu.tsx

```diff
@@ -24,6 +25,13 @@ export default function FlowMenu(): JSX.Element {
         (state) => state.flows.byId[state.flows.selected[0]],
     );
 
+    const selectedFlows = useAppSelector((state) => state.flows.selected);
+
+    const hasSingleFlowSelected = React.useMemo(
+        () => selectedFlows.length === 1,
+        [selectedFlows],
+    );
```

Comment:
I don't think memoization here is helping. We already get the forced rerender from `const selectedFlows = useAppSelector((state) => state.flows.selected);`. Memoizing here isn't going to change this. Two possible ways forward:

1. We decide that performance is good enough here. Then we can just remove the useMemo call and be done with it.
2. We decide that performance is important here. Then we need to get rid of the `useAppSelector((state) => state.flows.selected)` call. In this case we cannot call `removeFlow(selectedFlows)`. Alternatively, we could introduce an `removeSelectedFlows()` action. In this action we would then access the state to get the selected flows and work with that.

Both is fine with me, up to you what makes most sense. :)

Reply:
I would keep it simple and remove the **useMemo**. I don't think we will encounter any significant performance issues here. In most cases, users will select a relatively limited number of flows since they have to do it manually. I tested it with ~50 flows, and it runs smoothly. Does that make sense to you?

Reply:
SGTM

Reply:
perfect :)